src/containers/BooksForm.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+// Actions
+import { createBook } from '../actions/index';
+
+class BooksForm extends React.Component {
+  static getBookCategories() {
+    return [
+      'Action',
+      'Biography',
+      'History',
+      'Horror',
+      'Kids',
+      'Learning',
+      'Sci-Fi',
+    ];
+  }
+
+  constructor(props) {
+    super(props);
+
+    const defaultBookTitle = 'Book title';
+    const defaultBookCategory = BooksForm.getBookCategories()[0];
+
+    this.state = {
+      title: defaultBookTitle,
+      category: defaultBookCategory,
+    };
+
+    this.handleTitleChange = this.handleTitleChange.bind(this);
+    this.handleCategoryChange = this.handleCategoryChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit() {
+    const { createBook } = this.props;
+    createBook(this.state);
+  }
+
+  handleTitleChange(e) {
+    const { value } = e.target;
+    const newState = { title: value };
+
+    this.setState(newState);
+  }
+
+  handleCategoryChange(e) {
+    const { value } = e.target;
+    const newState = { category: value };
+
+    this.setState(newState);
+  }
```

Comment:
These function should be combined into one, because you shouldn't have a new function for every event that affects the state, it would make you component bloated.

Hint: use the same name for the state as the name of the dom element , then get the name from the DOM, `e.target` that way you know the exact element the event was called on and change that